docs/options.md

```diff
@@ -271,3 +271,41 @@ Valid options:
 | Default      | CLI Override                                                | API Override                                                |
 | ------------ | ----------------------------------------------------------- | ----------------------------------------------------------- |
 | `"preserve"` | <code>--prose-wrap <always&#124;never&#124;preserve></code> | <code>proseWrap: "<always&#124;never&#124;preserve>"</code> |
+
+## End of Line (EOL)
+
+_available in 1.15.0+_
+
+For historical reasons, there exist two commonly used flavors of line endings in text files. That is `\n` (or `LF` for _Line Feed_) and `\r\n` (or `CRLF` for _Carriage Return + Line Feed_).
+The former is common on Linux and macOS, while the latter is prevalent on Windows.
+Some details explaining why it is so [can be found on Wikipedia](https://en.wikipedia.org/wiki/Newline).
+
+By default, Prettier preserves a flavor of line endings a given file has already used.
+It also converts mixed line endings within one file to what it finds at the end of the first line.
+
+When people collaborate on a project from different operating systems, it becomes easy to end up with mixed line endings in the central git repository.
+It is also possible for Windows users to accidentally change line endings in an already committed file from `LF` to `CRLF`.
+Doing so produces a large `git diff`, and if it get unnoticed during code review, all line-by-line history for the file (`git blame`) gets lost.
+
+If you want to make sure that your git repository only contains Linux-style line endings in files covered by Prettier:
+
+1. Set `endOfLine` option to `lf`
+1. Configure [a pre-commit hook](./precommit.md) that will run Prettier
+1. Configure Prettier to run in your CI pipeline (e.g. using [`prettier-check` npm package](https://www.npmjs.com/package/prettier-check))
+1. Ask Windows users to run `git config core.autocrlf false` before working on your repo so that git did not convert `LF` to `CRLF` on checkout.
+   Alternatively, you can add `* text=auto eol=lf` to the repo's `.gitattributes` file to achieve this.
+
+All modern text editors in all operating systems are able to correctly display line endings when `\n` (`LF`) is used.
+However, old versions of Notepad for Windows will visually squash such lines into one.
+
+Valid options:
+
+- `"auto"` - Maintain existing line endings
+  (mixed values within one file are normalised by looking at what's used after the first line)
+- `"cr"` - Carriage Return character only (`\r`), used very rarely
```

Comment:
Is there a reason to support this value? I'd say we only support auto, crlf and lf.

Reply:
--> https://github.com/prettier/prettier/pull/5327#discussion_r230617150

Basically, it's for `.editorconfig` compatibility.

Reply:
The link no longer works because the discussion was marked as resolved. It's 4th in this review: https://github.com/prettier/prettier/pull/5327#pullrequestreview-171391657

Reply:
Thanks. It's sad but it looks we don't have a choice then.

Reply:
Can you put it at the end of the bulletted list since no one will likely want to use it?

Reply:
I'm tending to keep values ordered alphabetically both in the docs and in the code for consistency. Moving `cr` in the markdown would probably mean moving it in in `core-options.js` as well.

Perhaps we can re-order _all_ values except auto in the reverse alphabetical order, thus turning the list from `auto|cr|crlf|cr` into `auto|lf|crlf|cr` 🤔 That might be better actually! Shall I change?

Reply:
Marking this as resolved. See https://github.com/prettier/prettier/pull/5327#issuecomment-436055858 below.